gluonnlp/data/squad_dataset.py

```diff
@@ -0,0 +1,211 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=
+"""SQuAD dataset."""
+from mxnet import nd
+import numpy as np
+
+from gluonnlp import Vocab
+from gluonnlp.data.batchify import Pad
+
+__all__ = ['SQuAD', 'SQuADTransform']
+
+import io
+import json
+import os
+
+from mxnet.gluon.data import ArrayDataset
+from mxnet.gluon.utils import download, check_sha1, _get_repo_file_url
+from .registry import register
+
+
+@register(segment=['train', 'test'])
+class SQuAD(ArrayDataset):
+    """Stanford Question Answering Dataset (SQuAD) - reading comprehension dataset.
+
+    From
+    https://rajpurkar.github.io/SQuAD-explorer/
+
+    Parameters
+    ----------
+    segment : str, default 'train'
+        Dataset segment. Options are 'train' and 'dev'.
+    root : str, default '~/.mxnet/datasets/squad'
+        Path to temp folder for storing data.
+    """
+    def __init__(self, segment='train', root=os.path.join('~', '.mxnet', 'datasets', 'squad')):
+        self._data_file = {'train': ('train-v1.1.json',
+                                     '1faea1252438a64f9718412a55036b786cfcc636'),
+                           'dev': ('dev-v1.1.json',
+                                    'e1621aae0683b346ee9743bd5609266ba0cc34fc'),
+                           'word_vocab': ('word_vocab.json',
+                                   '13fd4dc500916612387c46fce46769638dccba5b'),
+                           'char_vocab': ('char_vocab.json',
+                                   'ca8fd555a45136e1643713041de53207c9aea2ad')}
```

Comment:
delete. this dataset class shouldn't provide anything that's not officially supplied.